packages/core/src/submodules/cbor/cbor-encode.ts

```diff
@@ -0,0 +1,250 @@
+import { fromUtf8 } from "@smithy/util-utf8";
+
+import {
+  CborMajorType,
+  extendedFloat16,
+  extendedFloat32,
+  extendedFloat64,
+  majorList,
+  majorMap,
+  majorNegativeInt64,
+  majorSpecial,
+  majorTag,
+  majorUint64,
+  majorUnstructuredByteString,
+  majorUtf8String,
+  specialFalse,
+  specialNull,
+  specialTrue,
+  Uint64,
+} from "./cbor-types";
+import { alloc } from "./cbor-types";
+
+const USE_BUFFER = typeof Buffer !== "undefined";
+
+type BufferWithUtf8Write = Buffer & {
+  utf8Write(str: string, index: number): number;
+};
+
+const initialSize = 2048;
+let data: Uint8Array = alloc(initialSize);
+let dataView: DataView = new DataView(data.buffer, data.byteOffset, data.byteLength);
+let cursor: number = 0;
+
+function ensureSpace(bytes: number) {
+  const remaining = data.byteLength - cursor;
+  if (remaining < bytes) {
+    if (cursor < 16_000_000) {
+      resize(Math.max(data.byteLength * 4, data.byteLength + bytes));
+    } else {
+      resize(data.byteLength + bytes + 16_000_000);
+    }
+  }
+}
+
+export function toUint8Array(): Uint8Array {
+  const out = alloc(cursor);
```

Comment:
I don't know of a specific Node.js API that maps to the term "buffer pooling". If you mean using for example creating a manager class that allocates and reclaims separate buffers, I don't believe it will improve the perf profile over using a single buffer, for reasons given below in the comment thread about the global buffer.

Initial development was done with individual buffers per encoding process, and iterative development showed that allocation/reclamation was costly and would unpredictably trigger GC. Even a well managed pool shouldn't beat the simplicity of a single buffer.